test/e2e/framework/framework.go

```diff
@@ -445,9 +454,47 @@ func (f *Framework) CreateNamespace(ctx context.Context, baseName string, labels
 	// fail to create serviceAccount in it.
 	f.AddNamespacesToDelete(ns)
 
+	if TestContext.DockerConfigFile != "" {
+		secret, err := f.CreateSecretFromDockerConfig(ns.Name)
+		ExpectNoError(err)
+
+		saPatch := patch{}
+		saPatch.ImagePullSecrets = append(saPatch.ImagePullSecrets, v1.LocalObjectReference{Name: secret.Name})
+
+		patchData, err := json.Marshal(saPatch)
+		if err != nil {
+			return ns, err
+		}
+
+		svc, err := f.ClientSet.CoreV1().ServiceAccounts(ns.Name).Patch(context.TODO(), serviceAccountName, types.StrategicMergePatchType, patchData, metav1.PatchOptions{})
+		if err != nil {
+			return ns, fmt.Errorf("[%s] Failed to patch imagePullSecrets to service account [%s]: %v", svc.Name, secret.Name, err)
+		}
+
+	}
+
 	return ns, err
 }
 
+// CreateSecretFromDockerConfig creates a secret using the private image registry credentials.
+// The credentials are provided by --docker-config-file flag.
+func (f *Framework) CreateSecretFromDockerConfig(namespace string) (*v1.Secret, error) {
+	contents, err := os.ReadFile(TestContext.DockerConfigFile)
+	ExpectNoError(err)
+	auth := string(contents)
+
+	secret := &v1.Secret{
+		Data: map[string][]byte{v1.DockerConfigJsonKey: []byte(auth)},
+		Type: v1.SecretTypeDockerConfigJson,
+	}
+	secret.Name = SecretName
+	Logf("create image pull secret %s", secret.Name)
+
+	_, err = f.ClientSet.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
```

Comment:
Don't use `context.TODO`. Instead pass in a ctx parameter to all functions that need it.